zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/ElasticsearchEngineClient.java

```diff
@@ -0,0 +1,183 @@
+
+    try {
+      client.cluster().putComponentTemplate(request);
+      log.debug(
+          "Component template [{}] was successfully created", templateDescriptor.getTemplateName());
+    } catch (final IOException e) {
+      log.error("Component template [{}] was NOT created", templateDescriptor.getTemplateName(), e);
+      throw new ExporterElasticsearchException(e.getMessage());
+    }
+  }
+
+  @Override
+  public void putMapping(final IndexDescriptor indexDescriptor, final String properties) {
+    final PutMappingRequest request = putMappingRequest(indexDescriptor, properties);
+
+    try {
+      client.indices().putMapping(request);
+      log.debug("Mapping in [{}] was successfully updated", indexDescriptor.getIndexName());
+    } catch (final IOException e) {
+      log.error("Mapping in [{}] was NOT updated", indexDescriptor.getIndexName(), e);
+      throw new ExporterElasticsearchException(e.getMessage());
+    }
+  }
+
+  private PutMappingRequest putMappingRequest(
+      final IndexDescriptor indexDescriptor, final String properties) {
+
+    try (final var propertiesStream = IOUtils.toInputStream(properties, StandardCharsets.UTF_8)) {
+      return new PutMappingRequest.Builder()
+          .index(indexDescriptor.getIndexName())
+          .properties(deserializeJson(TypeMapping._DESERIALIZER, propertiesStream).properties())
+          .build();
+    } catch (final IOException e) {
+      throw new RuntimeException(
+          "Failed to load properties json into stream for put mapping into indexes matching the descriptor: ["
+              + indexDescriptor
+              + "]",
+          e);
+    }
+  }
+
+  private PutComponentTemplateRequest putComponentTemplateRequest(
+      final ComponentTemplateDescriptor templateDescriptor) {
+    try (final var template =
+        getClass()
+            .getClassLoader()
+            .getResourceAsStream(templateDescriptor.getTemplateClasspathFileName())) {
+      return new PutComponentTemplateRequest.Builder()
+          .name(templateDescriptor.getTemplateName())
+          .withJson(template)
+          .create(templateDescriptor.create())
+          .build();
+    } catch (final IOException e) {
+      throw new RuntimeException(
+          "Failed to load json into stream for component template: ["
+              + templateDescriptor.getTemplateName()
+              + "]",
+          e);
+    }
+  }
+
+  public <T> T deserializeJson(final JsonpDeserializer<T> deserializer, final InputStream json) {
+    try (final var parser = client._jsonpMapper().jsonProvider().createParser(json)) {
+      return deserializer.deserialize(parser, client._jsonpMapper());
+    }
+  }
+
+  private InputStream descriptorMappings(final IndexDescriptor descriptor) {
+    //    return resourceRetriever.getResourceAsStream(descriptor.getMappingsClasspathFilename());
+    return getClass()
+        .getClassLoader()
+        .getResourceAsStream(descriptor.getMappingsClasspathFilename());
```

Comment:
we can extract this to a private method, then if we need to change it as suggested above, we can replace in one spot